diff --git a/SDL_FontCache.c b/SDL_FontCache.c
index 51acdb0..2f55bde 100644
--- a/SDL_FontCache.c
+++ b/SDL_FontCache.c
@@ -1110,17 +1110,8 @@ Uint8 FC_SetGlyphCacheLevel(FC_Font* font, int cache_level, FC_Image* cache_text
         // Grow cache?
         if(font->glyph_cache_count > font->glyph_cache_size)
         {
-            // Copy old cache to new one
-            int i;
-            FC_Image** new_cache;
-            new_cache = (FC_Image**)malloc(font->glyph_cache_count * sizeof(FC_Image*));
-            for(i = 0; i < font->glyph_cache_size; ++i)
-                new_cache[i] = font->glyph_cache[i];
-
-            // Save new cache
-            free(font->glyph_cache);
             font->glyph_cache_size = font->glyph_cache_count;
-            font->glyph_cache = new_cache;
+            font->glyph_cache = realloc(font->glyph_cache, font->glyph_cache_count * sizeof(FC_Image*));
         }
     }
 
